### ISSUE TITLE
Errors getting swallowed

### ISSUE THREAD

**Issue by djMax** [thebigredgeek/microlock#5]
If "other things" are wrong with etcd on a locking attempt, this handler will just eat them... Might be better to make sure it's the error we care about and rethrow if it isn't.

https://github.com/thebigredgeek/microlock/blob/master/src/microlock.js#L136

**Comment by thebigredgeek** [comment]
@djMax good idea.  Do you wanna put in a PR?  If not I will look into it later

**Comment by thebigredgeek** [comment]
@djMax bump

**Comment by djMax** [comment]
Haven't had time yet, but as we start to roll our use of it out of dev into production I'll definitely look at it.

**Comment by thebigredgeek** [comment]
@djMax have you guys rolled into prod yet?  Would love to collab on a PR within a couple of weeks, as I will have a bit more free time.  Currently all of my OSS time is consumed by `debug` :(

**Comment by djMax** [comment]
#6 addresses this.

**Comment by thebigredgeek** [comment]
@djMax are you sure tests were passing?  after I merged your PR, I can't seem to get integration tests passing :(

**Comment by djMax** [comment]
Yes, definitely. What fails for you?

*Sent with Shift
<https://tryshift.com/?utm_source=SentWithShift&utm_campaign=Sent%20with%20Shift%20Signature&utm_medium=Email%20Signature&utm_content=General%20Email%20Group>*

**Comment by djMax** [comment]
So this line:
```
          etcd.__setErr = true;
```

Means that the error captured is in fact "not ours". The test says "when unable to verify locked returns a rejecting promise passing AlreadyLockedError". Sending a PR with updated tests because I *think* that's the right move here.

**Comment by thebigredgeek** [comment]
Yup please update and then I'll release!  Thanks @djMax !
